components/formats-gpl/src/loci/formats/in/SlideBook6Reader.java

```diff
@@ -0,0 +1,395 @@
+	{
+		FormatTools.checkPlaneParameters(this, no, buf.length, x, y, w, h);
+
+		int[] zct = FormatTools.getZCTCoords(this, no);
+		int bpc = FormatTools.getBytesPerPixel(getPixelType());
+		byte[] b = new byte[FormatTools.getPlaneSize(this)];
+
+		readImagePlaneBuf(b, getSeries(), 0, zct[2], zct[0], zct[1]);
+
+		int pixel = bpc * getRGBChannelCount();
+		int rowLen = w * pixel;
+		for (int row=0; row<h; row++) {
+			System.arraycopy(b, pixel * ((row + y) * getSizeX() + x), buf,
+					row * rowLen, rowLen);
+		}
+
+		if (isRGB()) {
+			int bpp = getSizeC() * bpc;
+			int line = w * bpp;
+			for (int row=0; row<h; row++) {
+				for (int col=0; col<w; col++) {
+					int base = row * line + col * bpp;
+					for (int bb=0; bb<bpc; bb++) {
+						byte blue = buf[base + bpc*(getSizeC() - 1) + bb];
+						buf[base + bpc*(getSizeC() - 1) + bb] = buf[base + bb];
+						buf[base + bb] = blue;
+					}
+				}
+			}
+		}
+		return buf;
+	}
+
+	// -- Internal FormatReader API methods --
+	public void close(boolean fileOnly) throws IOException {
+		super.close(fileOnly);
+		if (libraryFound) {
+			closeFile();
+		}
+	}
+
+	/* @see loci.formats.FormatReader#initFile(String) */
+	protected void initFile(String id) throws FormatException, IOException {
+		super.initFile(id);
+
+		try {
+			openFile(id);
+
+			// read basic meta data
+			int numCaptures = getNumCaptures();
+			int[] numPositions = new int[numCaptures];
+			int[] numTimepoints = new int[numCaptures];
+			int[] numZPlanes = new int[numCaptures];
+			int[] numChannels = new int[numCaptures];
+			int[] numImages = new int[numCaptures];
+			for (int capture=0; capture < numCaptures; capture++) {
+				numPositions[capture] = getNumPositions(capture);
+				numTimepoints[capture] = getNumTimepoints(capture) / numPositions[capture];
+				numZPlanes[capture] = getNumZPlanes(capture);
+				numChannels[capture] = getNumChannels(capture);
+			}
+
+			core.clear();
+
+			// set up basic meta data
+			for (int capture=0; capture < numCaptures; capture++) {
+				CoreMetadata ms = new CoreMetadata();
+				core.add(ms);
+				setSeries(capture);
+				ms.sizeX = getNumXColumns(capture);
+				if (ms.sizeX % 2 != 0) ms.sizeX++;
+				ms.sizeY = getNumYRows(capture);
+				ms.sizeZ = numZPlanes[capture];
+				ms.sizeT = numTimepoints[capture] * numPositions[capture]; 
+				ms.sizeC = numChannels[capture];
+				int bytes = getBytesPerPixel(capture);
+				if (bytes % 3 == 0) {
+					ms.sizeC *= 3;
+					bytes /= 3;
+					ms.rgb = true;
+				}
+				else ms.rgb = false;
+
+				ms.pixelType = FormatTools.pixelTypeFromBytes(bytes, false, true);
+				ms.imageCount = ms.sizeZ * ms.sizeT;
+				if (!ms.rgb) 
+					ms.imageCount *= ms.sizeC;
+				ms.interleaved = true;
+				ms.littleEndian = true;
+				ms.dimensionOrder = "XYCZT";
+				ms.indexed = false;
+				ms.falseColor = false;
+			}
+			setSeries(0);
+
+			// fill in meta data
+			MetadataStore store = makeFilterMetadata();
+			MetadataTools.populatePixels(store, this);
```

Comment:
This line will still need to be updated before we can include the pull request - otherwise, all OME-XML generated via this reader will be invalid.